drivers/sensor/wsen_itds/itds.c

```diff
@@ -0,0 +1,380 @@
+		return -EINVAL;
+	}
+
+	*freq = itds_odr_map[idx].freq;
+	*mfreq = itds_odr_map[idx].mfreq;
+
+	return 0;
+}
+
+static int itds_accl_odr_set(struct device *dev, uint16_t freq, uint16_t mfreq)
+{
+	struct itds_device_data *ddata = dev->driver_data;
+	const struct itds_device_config *cfg = dev->config_info;
+	int start, end, i;
+	bool hp_mode;
+
+	hp_mode = !!(ddata->op_mode & ITDS_OP_MODE_HIGH_PERF);
+	if (hp_mode) {
+		start = ITDS_ODR_12_5;
+		end = ITDS_ODR_1600;
+	} else {
+		start = ITDS_ODR_1_6;
+		end = ITDS_ODR_200;
+	}
+
+	for (i = start; i <= end; i++) {
+		if ((freq == itds_odr_map[i].freq) &&
+		    (mfreq == itds_odr_map[i].mfreq)) {
+
+			return i2c_reg_update_byte(ddata->i2c, cfg->i2c_addr,
+					 ITDS_REG_CTRL1, ITDS_MASK_ODR, i << 4);
+		}
+	}
+
+	LOG_ERR("invalid odr, not in range");
+	return -EINVAL;
+}
+
+static int itds_accl_range_set(struct device *dev, int32_t range)
+{
+	struct itds_device_data *ddata = dev->driver_data;
+	const struct itds_device_config *cfg = dev->config_info;
+	int i, ret;
+	bool hp_mode;
+
+	for (i = 0; i < ITDS_ACCL_RANGE_END; i++) {
+		if (range <= (2 << i)) {
+			break;
+		}
+	}
+
+	if (i == ITDS_ACCL_RANGE_END) {
+		LOG_ERR("Accl out of range");
+		return -EINVAL;
+	}
+
+	ret = i2c_reg_update_byte(ddata->i2c, cfg->i2c_addr, ITDS_REG_CTRL6,
+				  ITDS_MASK_SCALE, i << 4);
+	if (ret) {
+		LOG_ERR("Accl set full scale failed %d", ret);
+		return ret;
+	}
+
+	hp_mode = !!(ddata->op_mode & ITDS_OP_MODE_HIGH_PERF);
+	ddata->scale = itds_sensitivity_scale[hp_mode][i];
+
+	return 0;
+}
+
+static int itds_attr_set(struct device *dev, enum sensor_channel chan,
+			 enum sensor_attribute attr,
+			 const struct sensor_value *val)
+{
+	if (chan != SENSOR_CHAN_ACCEL_X &&
+	    chan != SENSOR_CHAN_ACCEL_Y &&
+	    chan != SENSOR_CHAN_ACCEL_Z &&
+	    chan != SENSOR_CHAN_ACCEL_XYZ) {
+		LOG_ERR("attr_set() not supported on this channel.");
+		return -ENOTSUP;
+	}
+
+	switch (attr) {
+	case SENSOR_ATTR_FULL_SCALE:
+		return itds_accl_range_set(dev, sensor_ms2_to_g(val));
+
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		return itds_accl_odr_set(dev, val->val1, val->val2 / 1000);
+
+	default:
+		LOG_DBG("Accel attribute not supported.");
+		return -ENOTSUP;
+	}
+}
+
+static int itds_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct itds_device_data *ddata = dev->driver_data;
+	const struct itds_device_config *cfg = dev->config_info;
+	size_t i, ret;
+	uint8_t rval;
+
+	if (chan != SENSOR_CHAN_ACCEL_XYZ &&
+	    chan != SENSOR_CHAN_ACCEL_X &&
+	    chan != SENSOR_CHAN_ACCEL_Y &&
+	    chan != SENSOR_CHAN_ACCEL_Z &&
+	    chan != SENSOR_CHAN_DIE_TEMP) {
+		return -EINVAL;
+	}
+
+	ret = i2c_reg_read_byte(ddata->i2c, cfg->i2c_addr,
+				ITDS_REG_STATUS, &rval);
+	if (ret)
+		return ret;
+
+	if (!(rval & ITDS_EVENT_DRDY))
+		return -EAGAIN;
+
+	ret = i2c_burst_read(ddata->i2c, cfg->i2c_addr, ITDS_REG_X_OUT_L,
+			     (uint8_t *)ddata->samples,
+			     sizeof(uint16_t) * ITDS_SAMPLE_SIZE);
+	if (ret)
```

Comment:
All if/else should contain braces https://docs.zephyrproject.org/latest/contribute/index.html#coding-style